src/PrestaShopBundle/EventListener/Context/Admin/ShopContextListener.php

```diff
@@ -80,8 +89,10 @@ public function onKernelRequest(RequestEvent $event): void
         // In all cases a shop must be set for the context even if it's the default one
         if (!$shopConstraint->getShopId()) {
             $this->shopContextBuilder->setShopId((int) $this->configuration->get('PS_SHOP_DEFAULT', null, ShopConstraint::allShops()));
+            $this->shopContextAdapter->setShopContext((int) $this->configuration->get('PS_SHOP_DEFAULT', null, ShopConstraint::allShops()));
```

Comment:
The fallback should not be based only on the absence of shopId, the shop constraint has the real meaning Here you're missing the case for all shops and shop group And you're getting the same data twice which is not very efficient

Reply:
Besides the responsibility of filling the legacy context should not be handled by the listener IMHO but rather by the builder service We have two possibilities on where to do this:
- either in the setter as soon as they are called
- or in the build method, the risk being that if the new Context is never called then the fallback is never done either so for now using the setter is probably safer until the new Context is widely used and we're sure it can't be ignored

Reply:
ok, I think I misunderstood what was being done in the initshopcontext method, I'll look into it 👍  ok for the builder